src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -424,4 +495,31 @@ public boolean isInstantiable() {
         }
 
     }
+
+    private static class SlaveDisconnector extends MasterToSlaveCallable<Void, IOException> {
+
+        private static final long serialVersionUID = 8683427258340193283L;
+
+		private static final Logger LOGGER = Logger.getLogger(SlaveDisconnector.class.getName());
+
+        @Override
+        public Void call() throws IOException {
+            Engine e = Engine.current();
+            // No engine, do nothing.
+            if (e == null) {
+                return null;
+            }
+            // Ensure the slave supports setNoReconnect
+            try {
+                Engine.class.getMethod("setNoReconnect", boolean.class);
```

Comment:
The method has been there since 2009. I don't think this check is worth it. https://github.com/jenkinsci/remoting/commit/e43e787b5b31d9c4272135a731891a26f33ab0a4